src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp

```diff
@@ -231,12 +231,12 @@ XQSPIPS_Flasher::XQSPIPS_Flasher(std::shared_ptr<pcidev::pci_device> dev)
     // By default, it is 'perallel'
     mConnectMode = 0;
     // U30 Rev.A use single
-    if (typeStr.find("single")) {
+    if (typeStr.find("single") != std::string::npos) {
```

Comment:
shouldn't this needs to be if (typeStr.find("single") != 0)?

Reply:
I thought that's how "find" is used. Search for "single" until the end of the string

Reply:
But std::string::npos is -1, how does it match the original code here? @mamin506 , please confirm.

Reply:
typeStr.find("single") != std::string::npos is correct. If no matched was found, find() returns string::npos. The original code will failed if 'single' is the start of the string.